src/content/devices/input-output/js/test.js

```diff
@@ -0,0 +1,133 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+// This is a basic test file for use with testling.
+// The test script language comes from tape.
+/* jshint node: true */
+var test = require('tape');
+
+var webdriver = require('selenium-webdriver');
+var seleniumHelpers = require('../../../../../test/selenium-lib');
+
+test('Fake device selection and check video tag dimensions in ' +
+    'input-output demo', function(t) {
+  // FIXME: use env[SELENIUM_BROWSER] instead?
+  var driver = seleniumHelpers.buildDriver();
+
+  var webrtcDetectedBrowser = null;
+
+  driver.get('file://' + process.cwd() +
+      '/src/content/devices/input-output/index.html')
+  .then(function() {
+    t.pass('Page loaded');
+    return driver.executeScript('return webrtcDetectedBrowser;');
+  })
+  .then(function(browser) {
+    webrtcDetectedBrowser = browser;
+    // Making sure we can select the 1st audio device.
+    // TODO: Select more devices if Firefox adds a 2nd fake A&V device and
+    // Chrome adds another fake video device.
+    t.pass('Selecting 1st audio device');
+    return driver.findElement(webdriver.By.css('#audioSource>option'));
+  })
+  .then(function(element) {
+    return new webdriver.ActionSequence(driver).
+        doubleClick(element).perform();
+  })
+  // Check enumerateDevices has returned an id.
+  .then(function() {
+    return driver.findElement(webdriver.By.css(
+        '#audioSource>option')).getAttribute('value');
+  })
+  .then(function(deviceId) {
+    t.ok(deviceId !== '', 'Device/source id: ' + deviceId);
+  })
+  .then(function() {
+    // Making sure we can select the 1st video device.
+    // TODO: Select more devices if Firefox adds a 2nd fake A/V device and
+    // Chrome adds another fake video device.
+    t.pass('Selecting 1st video device');
+    return driver.findElement(webdriver.By.css('#videoSource>option'));
+  })
+  .then(function(element) {
+    return new webdriver.ActionSequence(driver).
+        doubleClick(element).perform();
+  })
+  // Check enumerateDevices has returned an id.
+  .then(function() {
+    return driver.findElement(webdriver.By.css(
+        '#videoSource>option')).getAttribute('value');
+  })
+  .then(function(deviceId) {
+    t.ok(deviceId !== '', 'Device/source id: ' + deviceId);
+  })
+  .then(function() {
+    // Make sure the stream is ready.
+    return driver.wait(function() {
+      return driver.executeScript('return window.stream !== undefined;');
+    }, 30 * 1000);
+  })
+   // Check for a fake audio device label (Chrome only).
```

Comment:
The problem is that the deviceId cannot be tied to the stream, sure you can test that after selecting a device that you get an active stream with appropriate tracks.

I want to verify that the device we select is the one used for the stream, having said that I see a flaw in my test, I do not select device based on the label, rather deviceId, this is due to it not being populated in the drop down, this could ofc be fixed by making sure real labels are populated in the drop down prior selecting them (I've seen fake labels being populated in Chrome while in selenium even though it's not served over HTTPS).